Remotion/Core/Development.Analyzers/RMCORE0001_SafeContextAnalyzer.cs

```diff
@@ -111,6 +123,89 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         match.CreateDiagnostic.Invoke(ctx, match.methodSymbol);
     }
 
+    private static void AnalyzeParallelInvocationExpression (
+        SyntaxNodeAnalysisContext ctx,
+        IReadOnlyCollection<(IMethodSymbol methodSymbol, Action<SyntaxNodeAnalysisContext, ISymbol> CreateDiagnostic)> methodSymbols,
+        INamedTypeSymbol? safeContextParallelTypeSymbol)
+    {
+      //Check whether the invocation expression should be covered
+      var invocationSymbolInfo = ctx.SemanticModel.GetSymbolInfo(ctx.Node);
+
+      if (invocationSymbolInfo.Symbol == null)
+        return;
+
+      var invocationOriginalDefinitionSymbol = invocationSymbolInfo.Symbol.OriginalDefinition;
+
+      var match = methodSymbols.FirstOrDefault(m => SymbolEqualityComparer.Default.Equals(m.methodSymbol, invocationOriginalDefinitionSymbol));
+      if (match == default)
```

Comment:
I suggest not using `default` for null checks. Yes it evaluates to `null` in this case but that is not clear without knowing the type of the variable.

I treat the `default` vs. `null` decision by considering if I care about the value or not. In this case, a null check, we do want to compare against `null` so using `null` makes the code more explicit. Using `default` indicates that the exact value is not relevant but a value is needed. A good example for this would be the `TryXXX` pattern with its `out` parameter that needs to be assigned even if the operation failed. In such a case, it makes sense to assign `default` as the value is meant to be irrelevant.